dump-scripts/python/zend_to_horizen.py

```diff
@@ -25,54 +45,93 @@
 	print(
-		"Usage: python3 {} <zend dump file name> <output_file>"
+		"Usage: python3 {} <zend dump file name> <mapping file name> <zend_vault_output_file> <eon_vault_automappings_file>"
 		.format(os.path.basename(__file__)))
 	sys.exit(1)
 
 zend_dump_file_name = sys.argv[1]
-result_file_name = sys.argv[2]
+mapped_addresses = {}
 
-total_balance = 0
+if len(sys.argv) == 3:
+	zend_vault_result_file_name = sys.argv[2]
+	eon_vault_result_file_name = None
+else:
+	mapping_file_name = sys.argv[2]
+	zend_vault_result_file_name = sys.argv[3]
+	eon_vault_result_file_name = sys.argv[4]
+	with open(mapping_file_name, 'r') as mapping_file:
+		mapped_addresses = json.load(mapping_file)
+
+total_balance_to_zend_vault = 0
+total_balance_to_eon_vault = 0
 total_balance_not_migrated = 0
 
 with open(zend_dump_file_name, 'r') as zend_dump_file:
 	zend_dump_data_reader = csv.reader(zend_dump_file)
 
-	results = {}
+	zend_vault_results = {}
+	eon_vault_results = {}
 
 	for (zend_address, balance_in_satoshi, _) in zend_dump_data_reader:
 		if not zend_address.startswith("unknown"):
-			try:
-				decoded_address = base58.b58decode_check(zend_address).hex()
-				# Remove prefix
-				decoded_address = "0x" + decoded_address[4:]
-				balance_in_wei = satoshi_2_wei(int(balance_in_satoshi))
-				total_balance = total_balance + balance_in_wei
-				if decoded_address in results:
-					print(
-						"Found 2 equal hashes: {0}, balance 1 {1}, balance 2 {2}"
-						.format(decoded_address, results[decoded_address], balance_in_wei ))
-					results[decoded_address] = results[decoded_address] + balance_in_wei
+			if int(balance_in_satoshi) != 0:
+				if zend_address in mapped_addresses:
+					balance_in_wei = satoshi_2_wei(int(balance_in_satoshi))
+					mapped_eth_address = mapped_addresses[zend_address].lower()
+					if mapped_eth_address in eon_vault_results:
+						eon_vault_results[mapped_eth_address] = eon_vault_results[mapped_eth_address] + balance_in_wei
+					else:
+						eon_vault_results[mapped_eth_address] = balance_in_wei
+					total_balance_to_eon_vault = total_balance_to_eon_vault + balance_in_wei
+					mapped_addresses.pop(zend_address)
 				else:
-					results[decoded_address] = balance_in_wei
-			except:
+					try:
+						decoded_address = base58.b58decode_check(zend_address).hex()
+						# Remove prefix
+						decoded_address = "0x" + decoded_address[4:]
+						balance_in_wei = satoshi_2_wei(int(balance_in_satoshi))
+						total_balance_to_zend_vault = total_balance_to_zend_vault + balance_in_wei
+						if decoded_address in zend_vault_results:
+							print(
+								"Found 2 equal hashes. Hash: {0}, balance 1: {1}, balance 2: {2}, current zend address: {3}"
+								.format(decoded_address, zend_vault_results[decoded_address], balance_in_wei, zend_address))
+							zend_vault_results[decoded_address] = zend_vault_results[decoded_address] + balance_in_wei
+						else:
+							zend_vault_results[decoded_address] = balance_in_wei
+					except:
```

Comment:
Maybe we could add some info to the error and also use an exception object, for instance:
```
except Exception as e:
    print(f"Error decoding address {zend_address}: {e}")
```

Reply:
Done